src/keylessBackup/hooks.ts

```diff
@@ -0,0 +1,150 @@
+import { KeylessBackupEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { valoraKeyshareIssued } from 'src/keylessBackup/slice'
+import { KeylessBackupFlow } from 'src/keylessBackup/types'
+import Logger from 'src/utils/Logger'
+import { PhoneNumberVerificationStatus } from 'src/verify/hooks'
+import networkConfig from 'src/web3/networkConfig'
+
+const TAG = 'keylessBackup/hooks'
+
+export function useVerifyPhoneNumber(phoneNumber: string, keylessBackupFlow: KeylessBackupFlow) {
+  const verificationCodeRequested = useRef(false)
+
+  const dispatch = useDispatch()
+
+  const [verificationStatus, setVerificationStatus] = useState(PhoneNumberVerificationStatus.NONE)
+  const [issueCodeCompleted, setIssueCodeCompleted] = useState(false)
+  const [smsCode, setSmsCode] = useState('')
+
+  // Async hook to make request to get sms code
+  useAsync(
+    async () => {
+      if (verificationCodeRequested.current) {
+        // verificationCodeRequested prevents the verification request from
+        // being fired multiple times, due to hot reloading during development
+        Logger.debug(
+          `${TAG}/issueSmsCode`,
+          'Skipping request to issueSmsCode since a request was already initiated'
+        )
+        return
+      }
+
+      ValoraAnalytics.track(KeylessBackupEvents.cab_issue_sms_code_start, { keylessBackupFlow })
+      Logger.debug(`${TAG}/issueSmsCode`, 'Initiating request')
+
+      const response = await fetch(networkConfig.cabIssueSmsCodeUrl, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          phoneNumber,
+          clientPlatform: Platform.OS,
+          clientBundleId: DeviceInfo.getBundleId(),
+        }),
+      })
+      if (response.ok) {
+        return response
+      } else {
+        throw new Error(await response.text())
+      }
+    },
+    [phoneNumber],
+    {
+      onError: (error: Error) => {
+        ValoraAnalytics.track(KeylessBackupEvents.cab_issue_sms_code_error, { keylessBackupFlow })
+        Logger.debug(`${TAG}/issueSmsCode`, 'Received error from issueSmsCode', error)
+        dispatch(showError(ErrorMessages.PHONE_NUMBER_VERIFICATION_FAILURE))
+      },
+      onSuccess: async (response?: Response) => {
+        if (!response) {
+          return
+        }
+        setIssueCodeCompleted(true)
+        verificationCodeRequested.current = true
+
+        ValoraAnalytics.track(KeylessBackupEvents.cab_issue_sms_code_success, { keylessBackupFlow })
+        Logger.debug(`${TAG}/issueSmsCode`, 'Successfully issued sms code')
+      },
+    }
+  )
+
+  // Async hook to post code and get keyshare once user types in code
+  useAsync(
+    async () => {
+      // add issueCodeCompleted to this hook, in case the SMS is received by the
+      // user before the successful response from issueSmsCode
```

Comment:
this is more critical in CPV than here because the verifySmsCode endpoint requires the verificationId from the first request, but I still retained it because there could be weird error states. For example, if the issueSmsCode request fails to complete, but the user still got a code, they could potentially see a success tick mark and a failure banner.

Reply:
the error banner will be on the code entry screen, they can enter the code but nothing will happen. Although if we removed this check, we can still attempt to validate and it may succeed. We could update it to behave this way though, but this would deviate from the CPV behavior. What do you think?

Reply:
I don't think there's a need to change it from its current behavior -- was just wondering!